swarm/storage/localstore/gc.go

```diff
@@ -85,6 +138,46 @@ func (db *DB) incGCSize(count int64) {
 	}
 }
 
+var writeGCSizeDelay = 10 * time.Second
+
+// writeGCSizeWorker calls writeGCSize function
+// on writeGCSizeTrigger receive. It implements a
+// backoff with delay of writeGCSizeDelay duration
+// to avoid very frequent database operations.
+func (db *DB) writeGCSizeWorker() {
+	for {
+		select {
+		case <-db.writeGCSizeTrigger:
+			err := db.writeGCSize()
+			if err != nil {
+				log.Error("localstore write gc size", "err", err)
+			}
+			select {
+			case <-time.After(writeGCSizeDelay):
+			case <-db.close:
+				return
+			}
+		case <-db.close:
+			return
+		}
+	}
+}
+
+// writeGCSize stores the number of items in gcIndex.
+// It removes all hashes from gcUncountedHashesIndex
+// not to include them on the next database initialization
+// when gcSize is counted.
+func (db *DB) writeGCSize() (err error) {
+	gcSize := atomic.LoadInt64(&db.gcSize)
+	err = db.storedGCSize.Put(uint64(gcSize))
```

Comment:
this should be batched with. thje deletes no?

Reply:
Updated.